agentos/registry.py

```diff
@@ -474,19 +436,47 @@ def add_run_artifacts(
         finally:
             shutil.rmtree(tmp_dir_path)
 
-    def add_run_command_spec(self, run_command_spec: RunCommandSpec) -> None:
+    def get_run(self, run_id: str) -> "Run":
+        from agentos.run import Run
+
+        return Run.from_registry(self, run_id)
+
+    def _request_spec_from_web_server(
+        self, spec_type: str, identifier: str, flatten: bool
+    ):
+        assert spec_type in ["run", "runcommand", "repo", "component"]
+        req_url = f"{self.root_api_url}/{spec_type}s/{identifier}/"
+        response = requests.get(req_url)
+        self._check_response(response)
+        flat_spec = json.loads(response.content)
+        print("spec returned was:")
+        print(flat_spec)
+        for spec_key, spec_val in [
+            (k, v)
+            for k, v in flat_spec.items()
+            if k.endswith("_link") or not v
```

Comment:
It might be useful to have a canonical set of keys for each spec type defined somewhere (maybe in `agentos/specs.py`?) and just drop any key not in the canonical set (and/or have some casting and validation functions built on top of the spec definitions).

Reply:
I've added classes in `agentos/specs.py` like `ComponentSpecKeys` which whose members are strings that represent the spec keys for each spec type. I'm going to leave this to a future PR which I've already started thinking about for #250